src/mame/drivers/spectrum.cpp

```diff
@@ -893,6 +893,24 @@ ROM_START(spectrum)
 	ROMX_LOAD("isomoje.rom",0x0000,0x4000, CRC(62ab3640) SHA1(04adbdb1380d6ccd4ab26ddd61b9ccbba462a60f), ROM_BIOS(18))
 	ROM_SYSTEM_BIOS(19, "iso8", "ISO ROM 8")
 	ROMX_LOAD("iso8bm.rom",0x0000,0x4000, CRC(43e9c2fd) SHA1(5752e6f789769475711b91e0a75911fa5232c767), ROM_BIOS(19))
+	ROM_SYSTEM_BIOS(20, "diagv51", "DiagROM v1.51")
+	ROMX_LOAD("diagrom.v51",0x0000,0x4000, CRC(83034df6) SHA1(e57b2c8a8e3563ea02a20eecd1d4cb6be9f9c2df), ROM_BIOS(20))
+	ROM_SYSTEM_BIOS(21, "diagv50", "DiagROM v1.50")
+	ROMX_LOAD("diagrom.v50",0x0000,0x4000, CRC(054a2d6a) SHA1(2fa6f3b31d7bb610df78be5c4c80537cf6e5911d), ROM_BIOS(21))
+	ROM_SYSTEM_BIOS(22, "diagv47", "DiagROM v1.47")
+	ROMX_LOAD("diagrom.v47",0x0000,0x4000, CRC(be329fc0) SHA1(24dc875753e729452f1ff941d2e0ae49f9031bc3), ROM_BIOS(22))
+	ROM_SYSTEM_BIOS(23, "diagv35", "DiagROM v1.35")
+	ROMX_LOAD("diagrom.v35",0x0000,0x4000, CRC(c869b642) SHA1(61eb786249a207ea57cdea6bb01fcd0b95c218f2), ROM_BIOS(23))
+	ROM_SYSTEM_BIOS(24, "diagv34", "DiagROM v1.34")
+	ROMX_LOAD("diagrom.v34",0x0000,0x4000, CRC(50478f29) SHA1(1e55af6e7bb24090ce052e7536e91f92492fe168), ROM_BIOS(24))
+	ROM_SYSTEM_BIOS(25, "diagv33", "DiagROM v1.33")
+	ROMX_LOAD("diagrom.v33",0x0000,0x4000, CRC(0fb080e1) SHA1(222f535ae277d0b8ce6c27b75319da9c45f8cadc), ROM_BIOS(25))
+	ROM_SYSTEM_BIOS(26, "diagv31", "DiagROM v1.31")
+	ROMX_LOAD("diagrom.v31",0x0000,0x4000, CRC(cddbb11e) SHA1(3ed203034fe1562ee7216fad06309d3f7033a967), ROM_BIOS(26))
+	ROM_SYSTEM_BIOS(27, "diagv28", "DiagROM v1.28")
+	ROMX_LOAD("diagrom.v28",0x0000,0x4000, CRC(aadf967e) SHA1(6fbdb11e475499655df36343a22f15d67f578165), ROM_BIOS(27))
```

Comment:
Please put these diagnostic ROMs in ascending order by version.  Other ROMs are in ascending version order here (BusySoft Upgrade and Turbo).